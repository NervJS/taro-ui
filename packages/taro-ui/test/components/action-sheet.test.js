import React from 'react'
import { mount } from 'enzyme'
import AtActionSheet from '../../lib/components/action-sheet'
import AtActionSheetItem from '../../lib/components/action-sheet/body/item'

describe('ActionSheet Component', () => {
  it('render initial ActionSheet', () => {
    const component = mount(
      <AtActionSheet>
        <AtActionSheetItem>按钮一</AtActionSheetItem>
        <AtActionSheetItem>按钮二</AtActionSheetItem>
      </AtActionSheet>
    )
    expect(component.find('taro-view-core')).toHaveLength(6)
    expect(
      component.find('.at-action-sheet__item').children().first().text()
    ).toBe('按钮一')
    expect(
      component.find('.at-action-sheet__item').children().last().text()
    ).toBe('按钮二')
    expect(component.html()).toMatchSnapshot()
  })

  it('render ActionSheet -- props title', () => {
    const component = mount(
      <AtActionSheet title='title'>
        <AtActionSheetItem>按钮一</AtActionSheetItem>
        <AtActionSheetItem>按钮二</AtActionSheetItem>
      </AtActionSheet>
    )
    expect(
      component.find('.at-action-sheet__header').children().first().text()
    ).toBe('title')
    expect(component.html()).toMatchSnapshot()
  })

  it('render ActionSheet -- props cancelText', () => {
    const component = mount(
      <AtActionSheet cancelText='cancel'>
        <AtActionSheetItem>按钮一</AtActionSheetItem>
        <AtActionSheetItem>按钮二</AtActionSheetItem>
      </AtActionSheet>
    )
    expect(
      component.find('.at-action-sheet__footer').children().first().text()
    ).toBe('cancel')
    expect(component.html()).toMatchSnapshot()
  })

  it('render ActionSheet -- props isOpened ', () => {
    const component = mount(
      <AtActionSheet isOpened>
        <AtActionSheetItem>按钮一</AtActionSheetItem>
        <AtActionSheetItem>按钮二</AtActionSheetItem>
      </AtActionSheet>
    )
    expect(
      component.find('.at-action-sheet--active').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })
})

// describe('ActionSheet Behavior ', () => {
//   it('ActionSheet onClose & onCancel & onClick', () => {
//     const onClose = jest.fn()
//     const onClick = jest.fn()
//     const onCancel = jest.fn()

//     const component = renderIntoDocument(
//       <AtActionSheet
//         isOpened
//         cancelText='取消'
//         onClose={onClose}
//         onCancel={onCancel}
//         title='清除位置信息后， 别人将不能查看到你\r\n可以通过转义字符换行'
//       >
//         <AtActionSheetItem onClick={onClick}>按钮一</AtActionSheetItem>
//       </AtActionSheet>
//     )
//     const componentDom = findDOMNode(component, 'at-action-sheet')

//     const bodyItemDom = componentDom.querySelector('.at-action-sheet__item')
//     const footerDom = componentDom.querySelector('.at-action-sheet__footer')
//     const overlayDom = componentDom.querySelector('.at-action-sheet__overlay')

//     Simulate.click(bodyItemDom)
//     expect(onClick).toBeCalled()

//     Simulate.click(footerDom)
//     expect(onCancel).toBeCalled()

//     Simulate.click(overlayDom)
//     process.nextTick(() => {
//       expect(onClose).toBeCalled()
//     })
//   })

//   it('ActionSheet onClose', () => {
//     const onClose = jest.fn()

//     const component = renderIntoDocument(
//       <AtActionSheet
//         isOpened
//         cancelText='取消'
//         onClose={onClose}
//         title='清除位置信息后， 别人将不能查看到你\r\n可以通过转义字符换行'
//       >
//         <AtActionSheetItem>按钮一</AtActionSheetItem>
//       </AtActionSheet>
//     )
//     const componentDom = findDOMNode(component, 'at-action-sheet')
//     const footerDom = componentDom.querySelector('.at-action-sheet__footer')

//     Simulate.click(footerDom)
//     process.nextTick(() => {
//       expect(onClose).toBeCalled()
//     })
//   })
// })
