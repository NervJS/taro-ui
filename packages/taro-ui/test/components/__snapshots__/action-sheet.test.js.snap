// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet Component render ActionSheet -- props cancelText 1`] = `"<taro-view-core class=\\"at-action-sheet\\"><taro-view-core class=\\"at-action-sheet__overlay\\"></taro-view-core><taro-view-core class=\\"at-action-sheet__container\\"><taro-view-core class=\\"at-action-sheet__body\\"><taro-view-core class=\\"at-action-sheet__item\\">按钮一</taro-view-core><taro-view-core class=\\"at-action-sheet__item\\">按钮二</taro-view-core></taro-view-core><taro-view-core class=\\"at-action-sheet__footer\\">cancel</taro-view-core></taro-view-core></taro-view-core>"`;

exports[`ActionSheet Component render ActionSheet -- props isOpened  1`] = `"<taro-view-core class=\\"at-action-sheet at-action-sheet--active\\"><taro-view-core class=\\"at-action-sheet__overlay\\"></taro-view-core><taro-view-core class=\\"at-action-sheet__container\\"><taro-view-core class=\\"at-action-sheet__body\\"><taro-view-core class=\\"at-action-sheet__item\\">按钮一</taro-view-core><taro-view-core class=\\"at-action-sheet__item\\">按钮二</taro-view-core></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`ActionSheet Component render ActionSheet -- props title 1`] = `"<taro-view-core class=\\"at-action-sheet\\"><taro-view-core class=\\"at-action-sheet__overlay\\"></taro-view-core><taro-view-core class=\\"at-action-sheet__container\\"><taro-view-core class=\\"at-action-sheet__header\\">title</taro-view-core><taro-view-core class=\\"at-action-sheet__body\\"><taro-view-core class=\\"at-action-sheet__item\\">按钮一</taro-view-core><taro-view-core class=\\"at-action-sheet__item\\">按钮二</taro-view-core></taro-view-core></taro-view-core></taro-view-core>"`;

exports[`ActionSheet Component render initial ActionSheet 1`] = `"<taro-view-core class=\\"at-action-sheet\\"><taro-view-core class=\\"at-action-sheet__overlay\\"></taro-view-core><taro-view-core class=\\"at-action-sheet__container\\"><taro-view-core class=\\"at-action-sheet__body\\"><taro-view-core class=\\"at-action-sheet__item\\">按钮一</taro-view-core><taro-view-core class=\\"at-action-sheet__item\\">按钮二</taro-view-core></taro-view-core></taro-view-core></taro-view-core>"`;
